core/hostservices.c

```diff
@@ -377,9 +377,35 @@ int hservice_send_error_log(uint32_t plid, uint32_t dsize, void *data)
 	return 0;
 }
 
-static int hservice_scom_read(uint64_t chip_id, uint64_t addr, void *buf)
+/* Convert a Opal XSCOM return code to one of the defined
+   return codes defined in the HBRT API. */
```

Comment:
@dcrowell77 IIUC you want opal-prd to return below error code to HBRT for scom errors. If yes, then this is not the place to fix. We need fix in external/opal-prd/opal-prd.c / hservice_scom_read() .. so that we return right values.

-Vasant

Reply:
Yes, that is the goal.  I wasn't sure where/when the PIB code got obfuscated so I picked what I thought was the most external spot.  Why are there 2 copies of the same function?  What is the version in core/hostservices.C used for?  Is it the old native (not opal-prd) support?

I'll push up a change that moves the logic over.

Reply:
Sorry. I missed to respond to this issue.

We have both user space and OPAL stuff in same repo.. anything under `external/opal-prd` is for user space. core/hostserives.c is for old native OPAL support.